src/animalShelter/AnimalShelter.ts

```diff
@@ -49,7 +52,7 @@ export default class AnimalShelter {
         return;
       }
 
-      if (animal.getType() === "Lion") {
+      if (animal instanceof Lion) {
         if (animal.getHome() !== undefined) {
           const family = families.find(
```

Comment:
can we somehow make the code that you dont have to search for the family and remove the animal explicitly when you want to remove the home from the animal (or change it). IF we do this in a couple of places in the code i think we will forget to do the whole sequence of code somewhere

Reply:
This will be addressed at a future task